tests/Tests/500__jupyterhub/test-jupyterlab-flask-notebook.robot

```diff
@@ -0,0 +1,85 @@
+Library          DebugLibrary
+Suite Setup      Begin Web Test
+Suite Teardown   End Web Test
+
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Library          DebugLibrary
+Library          JupyterLibrary
+
+
+*** Variables ***
+
+
+*** Test Cases ***
+Open RHODS Dashboard
+  [Tags]  Sanity
+  Wait for RHODS Dashboard to Load
+
+Can Launch Jupyterhub
+  [Tags]  Sanity
+  Launch JupyterHub From RHODS Dashboard Dropdown
+
+Can Login to Jupyterhub
+  [Tags]  Sanity
+  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  ${authorization_required} =  Is Service Account Authorization Required
+  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+
+Can Spawn Notebook
+  [Tags]  Sanity
+  Fix Spawner Status
+  Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook
+
+Can Run Flask Test Notebook
+  [Tags]  Sanity
+
+  Wait for JupyterLab Splash Screen  timeout=30
+
+  Maybe Select Kernel
+  ${is_launcher_selected} =  Run Keyword And Return Status  JupyterLab Launcher Tab Is Selected
+  Run Keyword If  not ${is_launcher_selected}  Open JupyterLab Launcher
+  Launch a new JupyterLab Document
+
+  Close Other JupyterLab Tabs
+
+  ##################################################
+  # Git clone repo and run existing notebook
+  ##################################################
+  Navigate Home (Root folder) In JupyterLab Sidebar File Browser
+  Open With JupyterLab Menu  Git  Clone a Repository
+  Input Text  //div[.="Clone a repo"]/../div[contains(@class, "jp-Dialog-body")]//input  https://github.com/tmckayus/flask-notebook
+  Click Element  xpath://div[.="CLONE"]
+
+  # Run the flask server
+  Open With JupyterLab Menu  File  Open from Path…
+  Input Text  //div[.="Open Path"]/../div[contains(@class, "jp-Dialog-body")]//input  flask-notebook/run_flask.ipynb
+  Click Element  xpath://div[.="Open"]
+  Wait Until run_flask.ipynb JupyterLab Tab Is Selected
+  Close Other JupyterLab Tabs
+  Open With JupyterLab Menu  Run  Run All Cells
+  Sleep  15
+
+  # Run the curl command
+  Open With JupyterLab Menu  File  Open from Path…
+  Input Text  //div[.="Open Path"]/../div[contains(@class, "jp-Dialog-body")]//input  flask-notebook/curl_local.ipynb
+  Click Element  xpath://div[.="Open"]
+  Wait Until curl_local.ipynb JupyterLab Tab Is Selected
+  Open With JupyterLab Menu  Run  Run All Cells
+  Sleep  15
+  ${output} =  Get Text  (//div[contains(@class,"jp-OutputArea-output")])[last()]
+  Should Match  ${output}  Hello World!
+  Close JupyterLab Selected Tab
+
+  # Shutdown flask server
+  Wait Until run_flask.ipynb JupyterLab Tab Is Selected
+  Open With JupyterLab Menu  Kernel  Interrupt Kernel
+  Wait Until JupyterLab Code Cell Is Not Active
+
+  Close JupyterLab Selected Tab
+  Logout JupyterLab
```

Comment:
This should not be needed since the Teardown will close the server

Reply:
I tried removing pieces of the shutdown code, anything less than this for some reason seems prone to failures in the teardown suite.  The screen never switches from displaying the activated shutdown button to having only "Start My Server".

Reply:
The only thing I'm not comfortable with is that Logout, because the Teardown then will not be able to go to the Hub Control Panel to shut down the server once you've logged out, so it will just close the browser without stopping the running server itself. If the test runs alone it's fine, but when there are multiple test cases being run in sequence this might cause some weird issues. Could you try removing it and checking if it still works?

Reply:
okay I removed the logout. The trouble is that the shutdown procedure seems to have trouble pressing the "stop server" button so the server never shuts down (it seems to if I press it myself for the robot). I'll keep playing.